weedcoco/tests/repo/test_deposit.py

```diff
@@ -0,0 +1,124 @@
+import pathlib
+import json
+import pytest
+import os
+import filecmp
+import re
+from weedcoco.repo.deposit import main
+from weedcoco.validation import ValidationError
+
+TEST_DATA_DIR = pathlib.Path(__file__).parent / "deposit_data"
+TEST_DATA_SAMPLE_DIR = pathlib.Path(__file__).parent / "deposit_data_sample"
+TEST_BASIC_DIR_1 = TEST_DATA_DIR / "basic_1"
+TEST_BASIC_DIR_2 = TEST_DATA_DIR / "basic_2"
+TEST_COMPLETE_DIR = TEST_DATA_DIR / "complete"
+TEST_DUPLICATE_DIR = TEST_DATA_DIR / "duplicate"
+
+
+@pytest.fixture
+def executor(tmpdir):
+    test_repo_dir = tmpdir / "test_repo"
+
+    class Executor:
+        def run(self, weedcoco_path, image_dir, repository_dir=test_repo_dir):
+            args = [
+                "--weedcoco-path",
+                weedcoco_path,
+                "--image-dir",
+                image_dir,
+                "--repository-dir",
+                repository_dir,
+            ]
+            args = [str(arg) for arg in args]
+            main(args)
+            return test_repo_dir
+
+    return Executor()
+
+
+def _cmp_files(dir1, dir2):
+
+    dirs_cmp = filecmp.dircmp(dir1, dir2, ignore=[".DS_Store"])
+    if (
+        len(dirs_cmp.left_only) > 0
+        or len(dirs_cmp.right_only) > 0
+        or len(dirs_cmp.funny_files) > 0
+    ):
+        print(dirs_cmp.left_only, dirs_cmp.right_only, dirs_cmp.funny_files)
+        return False
+    match, mismatch, errors = filecmp.cmpfiles(
+        dir1, dir2, dirs_cmp.common_files, shallow=False
+    )
+    if len(mismatch) > 0 or len(errors) > 0:
+        print(mismatch, errors)
+        return False
+    for common_dir in dirs_cmp.common_dirs:
+        new_dir1 = os.path.join(dir1, common_dir)
+        new_dir2 = os.path.join(dir2, common_dir)
+        if not _cmp_files(new_dir1, new_dir2):
+            return False
+    return True
+
+
+def _cmp_json(dir1, dir2):
+    def ordered_json(obj):
+        if isinstance(obj, dict):
+            return sorted((k, ordered_json(v)) for k, v in obj.items())
+        if isinstance(obj, list):
+            return sorted(ordered_json(x) for x in obj)
+        else:
+            return obj
+
+    return all(
+        [
+            ordered_json(json.loads(open(dir1 / dir / "weedcoco.json").read()))
```

Comment:
JSON can be parsed from a stream when not in a string:
```suggestion
            ordered_json(json.load(open(dir1 / dir / "weedcoco.json")))
```